include: zephyr: task_wdt: Easily disable Task Watchdog

RICCIARDI-Adrien opened the pull request:
The Task Watchdog can fire when a board is halted on a breakpoint during a debugging session.

These changes allow to easily neutralize the Task Watchdog by just setting `CONFIG_TASK_WDT=n` without having to comment all the Task Watchdog API function calls.

I used a `#ifdef` per function to avoid duplicating the comments.

RICCIARDI-Adrien commented:
OK, let me provide you a proposal.

RICCIARDI-Adrien committed:
task_wdt: Add dummy implementation

The introduced CONFIG_TASK_WDT_DUMMY Kconfig symbol allows to easily
disable the Task Watchdog by providing a dummy implementation.

The Task Watchdog can fire when a board is halted on a breakpoint during
a debugging session, so this new feature can turn useful in this case.

Signed-off-by: Adrien Ricciardi <aricciardi@baylibre.com>

martinjaeger reviewed (changes requested):
Thanks for pinging me and sorry for the late feedback. I had a look through the Zephyr code base and got the feeling that "dummy" instead of "test" would make it more clear that this implementation does nothing. So I'd suggest to rename the file and the Kconfig it to _dummy/_DUMMY. What do you think?

Otherwise looks good from my side.

RICCIARDI-Adrien commented:
I agree, this is more explicit with "dummy", I updated the PR accordingly.
Thanks for the various text improvements !

RICCIARDI-Adrien commented on subsys/task_wdt/CMakeLists.txt:
```diff
@@ -1,4 +1,12 @@
+    zephyr_sources(task_wdt_dummy.c)
+    message(WARNING "
+        Warning: the Dummy Task Watchdog is enabled and provides only a dummy
+        implementation.")
```
It is ! I updated the PR.

kartben reviewed (changes requested):
Please mention the new Kconfig in release notes for 4.3. thanks!

kartben commented on subsys/task_wdt/CMakeLists.txt:
```diff
@@ -1,4 +1,10 @@
-zephyr_sources_ifdef(CONFIG_TASK_WDT task_wdt.c)
+if (CONFIG_TASK_WDT_DUMMY)
+    zephyr_sources(task_wdt_dummy.c)
+    message(WARNING "Dummy Task Watchdog implementation is enabled.")
```
Agreed, it's just fine as-is

RICCIARDI-Adrien committed:
doc: releases: Add Dummy Task Watchdog Kconfig

Add the newly introduced Dummy Task Watchdog feature Kconfig to the 4.3
release notes.

Signed-off-by: Adrien Ricciardi <aricciardi@baylibre.com>

RICCIARDI-Adrien commented:
> Please mention the new Kconfig in release notes for 4.3. thanks!

Done !

RICCIARDI-Adrien commented:
Hi @martinjaeger and @kartben, could you approve the PR again ?

martinjaeger reviewed (approved):
Sorry for the late response. I was on holiday last week. Looks good, thanks.
